cranelift/wasm/src/code_translator.rs

```diff
@@ -1938,18 +1935,31 @@ pub fn translate_operator<FE: FuncEnvironment + ?Sized>(
 
             state.push1(builder.ins().snarrow(converted_a, zero));
         }
-        Operator::I32x4TruncSatF32x4U => {
+
+        // TODO: the relaxed instructions here are translated the same as the
+        // saturating instructions, even when the code generator configuration
+        // allow for different semantics across hosts. On x86, however, it's
+        // theoretically possible to have a slightly more optimal lowering which
+        // accounts for NaN differently, although the lowering is still not
+        // trivial (e.g. one instruction). At this time the
+        // more-optimal-but-still-large lowering for x86 is not implemented so
+        // the relaxed instructions are listed here instead of down below with
+        // the other relaxed instructions. An x86-specific implementation (or
+        // perhaps for other backends too) should be added and the codegen for
+        // the relaxed instruction should conditionally be different.
```

Comment:
This is a good note; can you create an issue to track this or let me know and I will create one? To me this comment boils down to "a better lowering is possible" and that seems like a good thing to remember to do.